src/Repository/NotificationDefinitionRepository.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Repository;
+
+use App\Client\Weather\OpenWeatherMap\Client;
+use App\Entity\NotificationDefinition;
+use App\Entity\User;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+/**
+ * @extends ServiceEntityRepository<NotificationDefinition>
+ *
+ * @method NotificationDefinition|null find($id, $lockMode = null, $lockVersion = null)
+ * @method NotificationDefinition|null findOneBy(array $criteria, array $orderBy = null)
+ * @method NotificationDefinition[]    findAll()
+ * @method NotificationDefinition[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class NotificationDefinitionRepository extends ServiceEntityRepository
+{
+    public function __construct(
+        ManagerRegistry $registry,
+        private Client $client,
+    ) {
+        parent::__construct($registry, NotificationDefinition::class);
+    }
+
+    public function save(NotificationDefinition $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->persist($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    public function remove(NotificationDefinition $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->remove($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    public function create(User $user, array $coordinates): NotificationDefinition
```

Comment:
Ta metoda powinna być w klasie `NotificationDefinition`, podobnie jak to masz z `Notification::fromDefinition()`. Repository powinno służyć tylko jako interfejs z bazą danych, czyli w tym przypadku jedyne co powinno zrobić repository, to zapisać stworzony w innym miejscu obiekt (metoda `save`).